src/ui/liferea_shell.c

```diff
@@ -569,10 +569,6 @@ on_notebook_scroll_event_null_cb (GtkWidget *widget, GdkEventScroll *event)
 static gboolean
 on_close (GtkWidget *widget, GdkEvent *event, gpointer user_data)
 {
-	guint signal_id = g_signal_lookup ("delete_event",  GTK_TYPE_WINDOW);
-
-	if (g_signal_has_handler_pending(widget, signal_id, (GQuark) 0, TRUE))
-		return FALSE;
```

Comment:
This does change the way the plugin works. Liferea shuts down instead of minimizing on close.

Reply:
@yaneti What was the intention of this removal? This is the last thing preventing the merge as I see it.

Reply:
@lwindolf the idea is to properly shutdown on window close button, regardless of the tray plugin state.  The current on_close code never does because on_close itself is still pending. 

I think I properly fixed the plugin after this removal in c78c0dd8abbbc9e712418cc880cbc42fc9e484f7  by blocking the main delete-event handler when the plugin is active.